### PR TITLE
otelcolconvert: support converting bearertoken and headerssetter auth

### LINKED ISSUE #6440: Support converting `bearertokenauthextension`

### LINKED ISSUE #6441: Support converting `headerssetterextension`

### PR DESCRIPTION
**Pull request by tpaschalis**
Closes #6440.
Closes #6441.

### CODE REVIEW

**Review by rfratto — COMMENTED** [review]
Nothing major, but a few small things before approving

File: internal/converter/internal/otelcolconvert/testdata/bearertoken.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+extensions:
+  bearertokenauth/fromfile:
+    token: "somerandomtoken"  # this will be ignored in lieu of the filename field.
+    filename: "file-containing.token"
```
[the remark this answers is not in the document]
Comment (rfratto) [reply internal/converter/internal/otelcolconvert/testdata/bearertoken.yaml:4]: Good to know :)

File: internal/converter/internal/otelcolconvert/testdata/bearertoken.river [hunk 2]
```diff
@@ -0,0 +1,38 @@
+local.file "default_fromfile" {
+	filename = "file-containing.token"
+}
+
+otelcol.auth.bearer "default_fromfile" {
+	token = "local.file.default_fromfile.content"
```
[the remark this answers is not in the document]
Comment (tpaschalis) [reply internal/converter/internal/otelcolconvert/testdata/bearertoken.river]: Fixed in https://github.com/grafana/agent/pull/6675/commits/7488979c1299001f21034516cd3dcb7e2d884b21

**Review by erikbaranowski — APPROVED** [review]
looks good. I removed 1 unused func param which is hopefully ok.

**Pull request merged**